src/dbms/inmemory/replication_handlers.cpp

```diff
@@ -1046,6 +1047,34 @@ uint64_t InMemoryReplicationHandlers::ReadAndApplyDeltas(storage::InMemoryStorag
             throw utils::BasicException("Failed to drop point index on :{}({})", data.label, data.property);
           }
         },
+        [&](WalVectorIndexCreate const &data) {
+          spdlog::trace("       Create vector index on :{}({})", data.label, data.property);
+          auto *transaction = get_replication_accessor(delta_timestamp, kUniqueAccess);
+          auto labelId = storage->NameToLabel(data.label);
+          auto propId = storage->NameToProperty(data.property);
+          auto const unum_metric_kind =
+              unum::usearch::metric_from_name(data.metric_kind.data(), data.metric_kind.size());
+          if (unum_metric_kind.error) {
+            throw utils::BasicException(
+                "Failed to create vector index on :{}({}), invalid metric kind: {}, supported metric kinds are: {}",
+                data.label, data.property, data.metric_kind, storage::SupportedMetricKindsToString());
+          }
+          auto const spec =
```

Comment:
I would prefer if shared_ptr was an internal detail.
Make the VectorIndexSpec and move it into CreateVectorIndex call, then eventually it can be moved into a shared ptr when it will be stored.